addons/udes_stock/models/stock_picking.py

```diff
@@ -739,3 +740,268 @@ def raise_stock_investigation(self, reason, quants, location):
+                    result_parent_package = Package.get_or_create(result_package_name, create=True)
+                # MPS: maybe this if is not needed
+                if not package:
+                    if product_ids:
+                        # Products are being packed
+                        result_package = Package.create({})
+                    elif not all([ml.result_package_id for ml in self]):
+                        # Setting result_parent_package expects to have
+                        # result_package for all the move lines
+                        raise ValidationError(
+                            _("Some of the move lines don't have result package.")
+                        )
+                    else:
+                        # We don't have either package or products and all lines have
+                        # result_package_id so parent_package should be result package parameter
+                        result_parent_package = Package.get_or_create(
+                            result_package_name, create=True
+                        )
+                        result_package = None
+                else:
+                    # Products are being packed into a new package
+                    result_package = None
+                    if product_ids:
+                        result_package = Package.create({})
+            # CASE C: wrong combination of package names given
+            elif product_ids:
+                raise ValidationError(
+                    _("Invalid parameters for target storage format, expecting result package.")
+                )
+
+        elif target_storage_format == "pallet_products":
+            if result_package_name:
+                # Moving stock into a pallet of products, result_package
+                # might be new pallet id
+                result_package = Package.get_or_create(result_package_name, create=True)
+                print("****** here we are")
+            elif product_ids and not result_package_name and not scan_parent_package_end:
+                raise ValidationError(
+                    _("Invalid parameters for target storage format, expecting result package.")
+                )
+
+        elif target_storage_format == "package":
+            if product_ids and not package and not result_package_name:
+                # Mark_as_done products without package or result_package
+                # Create result_package when packing products without
+                # result_package being set
+                result_package = Package.create({})
+
+        elif target_storage_format == "product":
+            # Error when trying to mark_as_done a full package or setting result package
+            # when result storage format is products
+            if result_package_name:
+                raise ValidationError(_("Invalid parameters for products target storage format."))
+
+        return result_package, result_parent_package
+
+    def picked(
```

Comment:
Should we give this a more expressive name, like `mark_picked_lines`?  `some_picking.picked()` doesn't tell the reader much about what might be happening.